src/librustdoc/html/static/main.js

```diff
@@ -1227,19 +1234,34 @@
             return [displayPath, href];
         }
 
+        function escape(content) {
+            var h1 = document.createElement('h1');
+            h1.textContent = content;
+            return h1.innerHTML;
+        }
+
+        function pathSplitter(path) {
+            var tmp = '<span>' + path.replace(/::/g, '::</span><span>');
+            if (tmp.endsWith("<span>")) {
+                return tmp.slice(0, tmp.length - 6);
+            }
+            return tmp;
+        }
+
         function addTab(array, query, display) {
             var extraStyle = '';
             if (display === false) {
                 extraStyle = ' style="display: none;"';
             }
 
             var output = '';
+            var duplicates = {};
```

Comment:
When can duplicates occur? Do you have some examples of when this is needed? The `shown` variable below is also trying to remove duplicates and I've tried to remove it in #50482 but if there are duplicates then we'll need to find a better way of dealing with them. `addTab` definitely isn't the correct place to be removing duplicates because it means the search result count will be incorrect: https://doc.rust-lang.org/nightly/std/?search=is_nan.

Reply:
It can occurs in case you have aliases. I need to fix the count then.